### PR TITLE
Hamza442/upgrade python version

### PR DESCRIPTION
**Pull request by Hamza442**
**Description:** Describe in a couple of sentences what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc.

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

File: .github/workflows/upgrade-python-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+     with:
+       branch: ${{ github.event.inputs.branch || '$default-branch' }}
+       team_reviewers: 'edx-data-engineering'
+       email_address: 'analytics@edx.opsgenie.net'
+       send_success_notification: true
+     secrets:
+       requirements_bot_github_token: ${{ secrets.REQUIREMENTS_BOT_GITHUB_TOKEN }}
+       requirements_bot_github_email: ${{ secrets.REQUIREMENTS_BOT_GITHUB_EMAIL }}
+       edx_smtp_username: ${{ secrets.EDX_SMTP_USERNAME }}
+       edx_smtp_password: ${{ secrets.EDX_SMTP_PASSWORD }}
```
Comment (jazibhumayun) [inline .github/workflows/upgrade-python-requirements.yml:27]: Please add an empty line at the end

File: requirements/base.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (jazibhumayun) [inline requirements/base.txt:2]: We use python3.8 venv in [jenkins schema builder job](http://jenkins-new.analytics.edx.org/view/Warehouse/job/snowflake-schema-builder/)

[https://github.com/edx/jenkins-job-dsl/blob/master/dataeng/jobs/analytics/SnowflakeSchemaBuilder.groovy#L48](https://github.com/edx/jenkins-job-dsl/blob/master/dataeng/jobs/analytics/SnowflakeSchemaBuilder.groovy#L48)

I am not sure how we should cater update in python version.

Comment (jazibhumayun) [reply requirements/base.txt:2]: There is also one concern that comes into my mind regarding python version update is dbt version's compatibility with the version we are updating.
I might happen when we update python version here whereas our current dbt version have some issue with that particular version

Comment (Hamza442) [reply requirements/base.txt:2]: Should we pin python version as well ?

Comment (jazibhumayun) [reply requirements/base.txt:2]: Yes, I think we should pin python version

**Pull request closed without merging**